src/constants/z-indices.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * This file cannot be converted to TypeScript because it's referenced in the
+ * Tailwind config.
+ */
+
+/**
+ * mapping of z-index names to z-indices; Ensure that these values are also
+ * written in the `tailwind.safelist.txt` file in the project root.
+ *
+ * @type {Record<string, string|number>}
+ */
+const Z_INDICES = {
+  auto: 'auto',
+  10: 10,
+  20: 20,
+  30: 30,
+  40: 40,
+  50: 50,
+  // Named indices
+  popover: 50,
+}
```

Comment:
This might as well be `const`. Typing it this way actually results in _less_ information for the type system as it collapses all the keys to `string`.

```suggestion
 * @type {const}
 */
const Z_INDICES = {
  auto: 'auto',
  10: 10,
  20: 20,
  30: 30,
  40: 40,
  50: 50,
  // Named indices
  popover: 50,
}
```

Though ideally it'd be frozen as well.

```suggestion
 */
const Z_INDICES = Object.freeze(/** @type {const} */ ({
  auto: 'auto',
  10: 10,
  20: 20,
  30: 30,
  40: 40,
  50: 50,
  // Named indices
  popover: 50,
}))
```